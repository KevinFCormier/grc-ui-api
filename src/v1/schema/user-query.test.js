/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('User Query Resolver', () => {
  test('Correctly Resolves Add User Query', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          mutation {
            saveQuery(resource: {name: "test", description: "test", searchText: "test"})
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Correctly Resolves Remove User Query', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          mutation {
            deleteQuery(resource: {name: "test", description: "test", searchText: "test"})
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });

  test('Correctly Resolves Get User Query', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          query userQueries {
            items: userQueries {
              name
              description
              searchText
            }
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
