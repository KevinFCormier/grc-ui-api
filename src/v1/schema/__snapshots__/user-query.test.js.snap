// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Query Resolver Correctly Resolves Add User Query 1`] = `
Object {
  "data": Object {
    "saveQuery": Object {
      "userQueries": Array [
        Object {
          "description": "test",
          "name": "test",
          "searchText": "test",
        },
      ],
    },
  },
}
`;

exports[`User Query Resolver Correctly Resolves Get User Query 1`] = `
Object {
  "data": Object {
    "items": Array [
      Object {
        "description": "test",
        "name": "test",
        "searchText": "test",
      },
    ],
  },
}
`;

exports[`User Query Resolver Correctly Resolves Remove User Query 1`] = `
Object {
  "data": Object {
    "deleteQuery": Object {
      "userQueries": Array [
        Object {
          "description": "test",
          "name": "test",
          "searchText": "test",
        },
      ],
    },
  },
}
`;
